weft/wefttest/wefttest.go

```diff
@@ -117,9 +121,63 @@ func (r Request) Do(server string) ([]byte, error) {
 		return nil, fmt.Errorf("%s got empty Content-Type header for response", r.ID)
 	}
 
+	if r.CSP != nil { //check response csp
+		cspString := res.Header.Get(("content-security-policy"))
+		if err := checkCSP(parseCspContent(cspString), r.CSP); err != nil {
+			//fmt.Println("## error URL", r.URL, cspString)
+			return nil, fmt.Errorf("%s got error content-security-policy header url %s, \n error: %s", r.ID, r.URL, err.Error())
+		}
+	}
+
 	return ioutil.ReadAll(res.Body)
 }
 
+/**
+ * convert CSP string to map
+ */
+func parseCspContent(content string) map[string]string {
+	cspArray := strings.Split(content, ";")
+	cspMap := make(map[string]string)
+	for _, csp := range cspArray {
+		s := strings.TrimSpace(csp)
```

Comment:
A single call to `strings.Fields()` can do this 5 lines trim-index thing.

Reply:
I think `strings.Fields()` works only for space, doesn't accept specific separator